compiler/AST/foralls.cpp

```diff
@@ -210,31 +210,131 @@ static ForallIntentTag argIntentToForallIntent(Expr* ref, IntentTag intent) {
   return TFI_DEFAULT; // dummy
 }
 
-ShadowVarSymbol* ShadowVarSymbol::buildFromArgIntent(IntentTag intent,
-                                                     Expr* ovar)
+static ShadowVarSymbol* buildShadowVariable(ShadowVarPrefix prefix,
+                                            const char* name, Expr* ovar)
 {
-  ForallIntentTag  fi = argIntentToForallIntent(ovar, intent);
-  const char*    name = toUnresolvedSymExpr(ovar)->unresolved;
-  return new ShadowVarSymbol(fi, name, NULL);
+  ForallIntentTag intent = TFI_DEFAULT; // dummy
+  switch (prefix)
+  {
+  case SVP_CONST:     intent = TFI_CONST;     break;
+  case SVP_IN:        intent = TFI_IN;        break;
+  case SVP_CONST_IN:  intent = TFI_CONST_IN;  break;
+  case SVP_REF:       intent = TFI_REF;       break;
+  case SVP_CONST_REF: intent = TFI_CONST_REF; break;
+  case SVP_VAR:
+    // This keyword is for a TPV.
+    // Whereas the user provided neither a type nor an init.
+    USR_FATAL_CONT(ovar, "a task private variable '%s'"
+                   "requires a type and/or initializing expression", name);
+    break;
+  }
+
+  ShadowVarSymbol* result = new ShadowVarSymbol(intent, name, NULL);
+  new DefExpr(result); // set result->defPoint
+  return result;
 }
 
+static ShadowVarSymbol* buildTaskPrivateVariable(ShadowVarPrefix prefix,
+            const char* nameString, Expr* nameExpr, Expr* type, Expr* init)
+{
+
+  // TPV - task-private variable, as we have a type and/or an initializer.
+  ShadowVarSymbol* result = new ShadowVarSymbol(TFI_TASK_PRIVATE,
+                                                nameString, NULL);
+
+  switch (prefix)
+  {
+  case SVP_VAR:        result->qual = QUAL_VAL;       break;
```

Comment:
I generally expect { to increase indent level, is there a particular reason the contents of the switch is not indented?